app/services/accept_proposed_organisation.rb

```diff
@@ -6,32 +6,61 @@ class Response
     NO_EMAIL = "No email"
     OTHER_ERROR = "Other Error"
 
-    attr_reader :status, :error_message, :accepted_organisation
+    attr_reader   :status, :error_message 
+    attr_accessor :accepted_org, :not_accepted_org
 
     def initialize status, error_message, accepted_organisation
       @status = status
       @error_message = error_message
-      @accepted_organisation = accepted_organisation
+      @accepted_org = accepted_organisation
+      @not_accepted_org = nil
     end
   end
+  
+  class NotificationError < StandardError; end
 
   def initialize(proposed_org)
     @proposed_org = proposed_org
     @email = @proposed_org.email
   end
 
   def run
-    org = @proposed_org.accept_proposal
-    usr = User.find_by(email: @email)
-    if usr
-      NotifyRegisteredUserFromProposedOrg.new(usr,org).run
-      Response.new(Response::NOTIFICATION_SENT,nil, org)
-    else
-      create_invitation_response_object(InviteUnregisteredUserFromProposedOrg.new(@email,org).run, org)
-    end
+    accept_organisation
+  rescue NotificationError
+    rollback_changes_to_org
   end
 
   private
+  
+  def accept_organisation
+    org = @proposed_org.accept_proposal; @result = inform_user org
+    raise NotificationError if success?(@result.status) == false
```

Comment:
when it comes to predicate method (ends with ? like `success?`) you don't need to compare with boolean for example we can change the expression to:
```ruby
raise NotificationError unless success?(@result.status) 
```
Usually I avoid using exception for managing the flow, it's ok to use it but as alternative I would check the status of `inform_user` before accepting the organisation.